packages/flutter_tools/test/general.shard/runner/runner_test.dart

```diff
@@ -43,17 +40,20 @@ void main() {
       // runner.run() asynchronously calls the exit function set above, so we
       // catch it in a zone.
       unawaited(runZoned<Future<void>>(
-        () {
-          unawaited(runner.run(
-            <String>['test'],
-            <FlutterCommand>[
-              CrashingFlutterCommand(),
-            ],
-            // This flutterVersion disables crash reporting.
-            flutterVersion: '[user-branch]/',
-            reportCrashes: true,
-          ));
-          return null;
+          () {
+            unawaited(runner.run(
+              <String>['test'],
+              <FlutterCommand>[
+                CrashingFlutterCommand(),
+              ],
+              // This flutterVersion disables crash reporting.
+              flutterVersion: '[user-branch]/',
+              reportCrashes: true,
+              overrides: <Type, Generator>{
+                FileSystem: () => MemoryFileSystem(),
```

Comment:
I know the double override is kind of weird, but I couldn't think of another way to do this so:
1. With a real crash, it always uses LocalFileSystem regardless of context (see previous note about recording):
```dart
/// File system used by the crash reporting logic.	
///	
/// We do not want to use the file system stored in the context because it may	
/// be recording. Additionally, in the case of a crash we do not trust the	
/// integrity of the [AppContext].
```
2. Still lets the test inject a MemoryFileSystem.

Reply:
We don't support recording anymore. The latter part about not trusting AppContext is a bit suspect, I don't really think we've had any issues for the last few years regarding AppContext